code/game/objects/items/storage/holsters.dm

```diff
@@ -42,19 +45,25 @@
 
 /obj/item/storage/holster/handle_item_insertion(obj/item/W, prevent_warning = 0)
 	. = ..()
-	if (!. || !(W.type in holsterable_allowed) ) //check to see if the item being inserted is the snowflake item
+	if(!. || !is_type_in_list(W,holsterable_allowed)) //check to see if the item being inserted is the snowflake item
 		return
 	holstered_item = W
 	playsound(src, sheathe_sound, 15, 1)
-	update_icon()
+	if(istype(src, /obj/item/storage/holster/belt)) //Belt holsters have a pistol-in-belt sprite to overlay over the belt
+		update_gun_icon()
+	else
+		update_icon()
 
 /obj/item/storage/holster/remove_from_storage(obj/item/W, atom/new_location, mob/user)
 	. = ..()
-	if (!. || !(W.type in holsterable_allowed) ) //check to see if the item being removed is the snowflake item
+	if(!. || !is_type_in_list(W,holsterable_allowed)) //check to see if the item being removed is the snowflake item
 		return
 	holstered_item = null
 	playsound(src, draw_sound, 15, 1)
-	update_icon()
+	if(istype(src, /obj/item/storage/holster/belt)) //Belt holsters have a pistol-in-belt sprite to overlay over the belt
+		update_gun_icon()
+	else
+		update_icon()
```

Comment:
Various stuff in update_gun_icon is already done by update_icon_state. It would probably better to always call update_icon, and just have the update_gun_icon behind an if statement. That way you can purge all the duplicate code out of update_gun_icon.